source/ColumnChooserPanel.cpp

```diff
@@ -0,0 +1,129 @@
+/* ColumnChooserPanel.cpp
+Copyright (c) 2024 by Endless Sky contributors
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ColumnChooserPanel.h"
+
+#include "FillShader.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "GameData.h"
+#include "Information.h"
+#include "Interface.h"
+#include "Preferences.h"
+#include "Sprite.h"
+#include "SpriteSet.h"
+#include "SpriteShader.h"
+#include "UI.h"
+#include "text/WrappedText.h"
+
+#include <numeric>
+
+using namespace std;
+
+ColumnChooserPanel::ColumnChooserPanel(const vector<PlayerInfoPanel::SortableColumn> &columns,
+	InfoPanelState *panelState)
+	: columns(columns), panelState(panelState)
+{
+	SetInterruptible(false);
+}
+
+
+
+void ColumnChooserPanel::Draw()
+{
+	if(GetUI()->IsTop(this))
+		DrawBackdrop();
+
+	zones.clear();
+
+	Information info;
+	const Interface *columnChooser = GameData::Interfaces().Get("columns menu");
+	info.SetCondition("columns menu open");
+	columnChooser->Draw(info, this);
+
+	const Font &font = FontSet::Get(14);
+	const Color * const dim = GameData::Colors().Get("dim");
+	const Color * const medium = GameData::Colors().Get("medium");
+	const Color * const bright = GameData::Colors().Get("bright");
+	const Sprite *box[] = {SpriteSet::Get("ui/unchecked"), SpriteSet::Get("ui/checked")};
+
+	Point topLeft(270., -280.);
+	const Point boxSize = Point(box[0]->Width(), box[0]->Height());
+	static const Point ROW_ADVANCE(0., 20.);
+	const set<const string> visibleColumns = panelState->VisibleColumns();
+	auto isVisible = [&](string name){ return visibleColumns.find(name) != visibleColumns.end(); };
```

Comment:
```suggestion
	auto isVisible = [&visibleColumns](string &name){ return visibleColumns.find(name) != visibleColumns.end(); };
```

Reply:
This is not necessary; a lambda already closes over any used variables.

Reply:
code clarity, I guess (see https://github.com/endless-sky/endless-sky/pull/9112#discussion_r1352316001)

Reply:
Hmm, I would recommend reverting those suggested changes by quyykk. Simpler code is easier to read and easier to maintain.
We ought to use the C++ default behaviour here as we do everywhere else (e.g. not setting auto-initialised variables to the same value they would get auto-initialised to).
It also makes refactoring easier as you have fewer places to e.g. change a variable name.

Reply:
I'm not against it, but I'd wait for others' opinions